tools/build_defs/repo/git.bzl

```diff
@@ -15,12 +15,84 @@
 
 load("@bazel_tools//tools/build_defs/repo:utils.bzl", "patch", "update_attrs", "workspace_and_buildfile")
 
+
+def _if_debug(cond, st, what="Action"):
+    "Print if 'cont'."
+    if cond:
+        print("{} returned {}\n{}\n----\n{}".format(what, st.return_code, st.stdout, st.stderr))
+
+def _hash(ctx, value):
+    """Hash a value... TODO needs a proper implementation.
+    """
+    bash_exe = ctx.os.environ["BAZEL_SH"] if "BAZEL_SH" in ctx.os.environ else "bash"
+    key = ctx.execute([
+        bash_exe,
+        "-c",
+        "(echo '{value}' | shasum)".format(value=value),
+    ]).stdout
+    key = key.split(sep=' ')[0]
+    return key
+
+def _setup_cache(ctx, git_cache):
+    # Set-up git_cache git repository.
+    bash_exe = ctx.os.environ["BAZEL_SH"] if "BAZEL_SH" in ctx.os.environ else "bash"
+    st = ctx.execute([bash_exe, "-c", """set -ex
+  mkdir -p {git_cache}
+  git -C {git_cache} init --bare || :
+  """.format(git_cache=git_cache)], environment = ctx.os.environ)
+    _if_debug(cond=ctx.attr.verbose, st=st, what='Init')
+    if st.return_code:
+        fail("Error ... {}:\n{}\n---\n{}".format(ctx.name, st.stdout, st.stderr))
+
+def _get_repository_from_cache(ctx, directory, ref, shallow, git_cache):
+    bash_exe = ctx.os.environ["BAZEL_SH"] if "BAZEL_SH" in ctx.os.environ else "bash"
+    exec_result = ctx.execute([bash_exe, "-c", """
+  cd {working_dir}
+  set -ex
+      rm -rf '{directory}' '{dir_link}'
+      git -C {git_cache} worktree prune
+      git -C {git_cache} worktree add '{directory}' {ref}
+      #git -C '{directory}' reset --hard {ref}
```

Comment:
Could you please clean up these two commented lines? They are not necessary because I ended up removing `{directory}` and doing a clean checkout of a worktree.